central/pruning/pruning.go

```diff
@@ -567,6 +573,45 @@ func (g *garbageCollectorImpl) removeOldReportHistory(config *storage.PrivateCon
 	}
 }
 
+func (g *garbageCollectorImpl) removeOldReportBlobs(config *storage.PrivateConfig) {
+	blobRetentionDays := config.GetReportRetentionConfig().GetDownloadableReportRetentionDays()
+	cutOffTime, err := types.TimestampProto(time.Now().Add(-time.Duration(blobRetentionDays) * 24 * time.Hour))
+	if err != nil {
+		log.Errorf("Failed to determine downloadable report retention %v", err)
+		return
+	}
+
+	// Sort reversely by modification time
+	query := search.NewQueryBuilder().AddRegexes(search.BlobName, common.ReportBlobRegex).WithPagination(
+		search.NewPagination().AddSortOption(search.NewSortOption(search.BlobModificationTime).Reversed(true)))
```

Comment:
Why are you order the blobs last modified (latest) first?

Reply:
From the latest, I took blobs and counting them until size quota filled or cut off time reached, then kill the rest.